iceoryx_posh/include/iceoryx_posh/internal/popo/base_client.inl

```diff
@@ -0,0 +1,204 @@
+{
+    return m_port.getUniqueID();
+}
+
+template <typename Port>
+inline capro::ServiceDescription BaseClient<Port>::getServiceDescription() const noexcept
+{
+    return m_port.getCaProServiceDescription();
+}
+
+template <typename Port>
+inline void BaseClient<Port>::connect() noexcept
+{
+    m_port.connect();
+}
+
+template <typename Port>
+inline ConnectionState BaseClient<Port>::getConnectionState() const noexcept
+{
+    return m_port.getConnectionState();
+}
+
+template <typename Port>
+inline void BaseClient<Port>::disconnect() noexcept
+{
+    m_port.disconnect();
+}
+
+template <typename Port>
+inline bool BaseClient<Port>::hasResponses() const noexcept
+{
+    return m_port.hasNewResponses();
+}
+
+template <typename Port>
+inline bool BaseClient<Port>::hasMissedResponses() noexcept
+{
+    return m_port.hasLostResponsesSinceLastCall();
+}
+
+template <typename Port>
+inline void BaseClient<Port>::releaseQueuedResponses() noexcept
+{
+    m_port.releaseQueuedResponses();
+}
+
+template <typename Port>
+inline void BaseClient<Port>::invalidateTrigger(const uint64_t uniqueTriggerId) noexcept
+{
+    if (m_trigger.getUniqueId() == uniqueTriggerId)
+    {
+        m_port.unsetConditionVariable();
+        m_trigger.invalidate();
+    }
+}
+
+template <typename Port>
+inline void BaseClient<Port>::enableState(iox::popo::TriggerHandle&& triggerHandle,
+                                          const ClientState clientState) noexcept
+{
+    switch (clientState)
+    {
+    case ClientState::HAS_RESPONSE:
+        if (m_trigger)
+        {
+            LogWarn()
+                << "The client is already attached with either the ClientState::HAS_RESPONSE or "
+                   "ClientEvent::RESPONSE_RECEIVED to a WaitSet/Listener. Detaching it from previous one and "
+                   "attaching it to the new one with ClientState::HAS_RESPONSE. Best practice is to call detach first.";
+
+            errorHandler(
+                Error::
+                    kPOPO__BASE_CLIENT_OVERRIDING_WITH_STATE_SINCE_HAS_RESPONSE_OR_RESPONSE_RECEIVED_ALREADY_ATTACHED,
+                nullptr,
+                ErrorLevel::MODERATE);
+        }
+        m_trigger = std::move(triggerHandle);
+        m_port.setConditionVariable(*m_trigger.getConditionVariableData(), m_trigger.getUniqueId());
+        break;
+    }
+}
+
+template <typename Port>
+inline WaitSetIsConditionSatisfiedCallback
+BaseClient<Port>::getCallbackForIsStateConditionSatisfied(const ClientState clientState) const noexcept
+{
+    switch (clientState)
+    {
+    case ClientState::HAS_RESPONSE:
+        return {*this, &SelfType::hasResponses};
+    }
+    return {};
+}
+
+template <typename Port>
+inline void BaseClient<Port>::disableState(const ClientState clientState) noexcept
+{
+    switch (clientState)
+    {
+    case ClientState::HAS_RESPONSE:
+        m_trigger.reset();
+        m_port.unsetConditionVariable();
+        break;
```

Comment:
Will there be a static code analysis finding if there is no default case?

Reply:
I'm not sure, but I would advocate to make exceptions for `enum class` if there is a return statement right after the `switch`. A default case makes it harder to extend the enum and find all locations where it must be added. The default state might be there to appease the tool or because one really does not want to handle a case. Without a default case, the compiler will complain and our CI would not allow to merge the PR.